tcms/rpc/tests/test_testplan.py

```diff
@@ -226,3 +222,114 @@ def test_ignores_existing_mappings(self):
+        with self.assertRaisesRegex(XmlRPCFault, 'This field is required.'):
+            self.rpc_client.TestPlan.create({
+                'product': product.pk,
+                'product_version': version.pk,
+                'name': '',
+                'type': plan_type.pk,
+                'text': 'Testing TCMS',
+                'parent': None,
+            })
+
+
+class TestCreatePermission(APIPermissionsTestCase):
+    permission_label = 'testplans.add_testplan'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.product = ProductFactory()
+        self.version = VersionFactory(product=self.product)
+        self.plan_type = PlanTypeFactory()
+        self.params = {
+            'product': self.product.pk,
+            'product_version': self.version.pk,
+            'name': 'Testplan foobar',
+            'type': self.plan_type.pk,
+            'text': 'Testing TCMS',
+            'parent': None,
+        }
+
+    def verify_api_with_permission(self):
+        result = self.rpc_client.TestPlan.create(self.params)
+        self.assertEqual(self.params['product'], result['product_id'])
+        self.assertEqual(self.params['product_version'], result['product_version_id'])
+        self.assertEqual(self.params['name'], result['name'])
+        self.assertEqual(self.params['type'], result['type_id'])
+        self.assertEqual(self.params['text'], result['text'])
+        self.assertEqual(self.params['parent'], result['parent'])
+
+        # verify object from DB
+        testplan = TestPlan.objects.get(name=self.params['name'])
+        self.assertEqual(testplan.serialize(), result)
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.TestPlan.create(self.params)
+
+
+class TestListAttachments(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.plan = TestPlanFactory()
+
+    def test_list_attachments(self):
+        file_name = 'attachment.txt'
+        self.rpc_client.TestPlan.add_attachment(self.plan.pk, file_name, 'a2l3aXRjbXM=')
+        attachments = self.rpc_client.TestPlan.list_attachments(self.plan.pk)
+        self.assertEqual(1, len(attachments))
+
+    def test_list_attachments_with_wrong_plan_id(self):
+        with self.assertRaisesRegex(XmlRPCFault, 'TestPlan matching query does not exist'):
+            self.rpc_client.TestPlan.list_attachments(-1)
+
+
+class TestListAttachmentsPermissions(APIPermissionsTestCase):
+    permission_label = 'attachments.view_attachment'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.plan = TestPlanFactory()
+
+    def verify_api_with_permission(self):
+        attachments = self.rpc_client.TestPlan.list_attachments(self.plan.pk)
+        self.assertEqual(0, len(attachments))
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegexp(ProtocolError, '403 Forbidden'):
```

Comment:
pylint says `Using deprecated method assertRaisesRegexp() (deprecated-method)`. Drop the `p` at the end.